plugins/html-plugin/src/main/java/ru/sbtqa/tag/pagefactory/find/HtmlFindUtils.java

```diff
@@ -0,0 +1,382 @@
+            } else {
+                errorMessage.append(currenElementName);
+            }
+            errorMessage.append(". The search was performed by path: ").append(element.getFullElementPath());
+        } else {
+            errorMessage.append(currenElementName);
+        }
+        return errorMessage.toString();
+    }
+
+    /**
+     * Searches for an element of the specified type by the specified name or
+     * path
+     * <p>
+     * As a path, an enumeration of elements in a nested structure can be used
+     * through the separator "{@code ->}", for example:
+     * <ul>
+     * <li> element in the block: Block{@code ->}Element
+     * <li> element in the list with the sequence number 3: List{@code ->}3
+     * </ul>
+     * <p>
+     * If an element on the page is declared as {@code WebElement}, then an
+     * attempt will be made to convert it to the passed type if this is possible
+     * <p>
+     * By default, expects an element to appear on the page
+     *
+     * @param <T> expected element type
+     * @param name element name or path
+     * @param type element type
+     * @return Returns an element from a page by name or path
+     */
+    public <T extends WebElement> T find(String name, Class<T> type) {
+        return find(name, type, true);
+    }
+
+    /**
+     * Searches for an element of the specified type by the specified name or
+     * path
+     * <p>
+     * As a path, an enumeration of elements in a nested structure can be used
+     * through the separator "{@code ->}", for example:
+     * <ul>
+     * <li> element in the block: Block {@code ->} Element
+     * <li> element in the list with the sequence number 3: List {@code ->} 3
+     * </ul>
+     * <p>
+     * If an element on the page is declared as {@code WebElement}, then an
+     * attempt will be made to convert it to the passed type if this is possible
+     *
+     * @param <T> expected element type
+     * @param name element name or path
+     * @param type element type
+     * @param wait to wait for the element to appear or not
+     * @return Returns an element from a page by name or path
+     */
+    public <T extends WebElement> T find(String name, Class<T> type, boolean wait) {
+        Class instanсeType = type;
+        T element = find(name, wait);
+        Class elementType = element.getClass();
+
+        if (isTypifiedElement(elementType) || isHtmlElement(elementType)) {
+            if (!type.isAssignableFrom(elementType)) {
+                throw new IncorrectElementTypeException(format("Found component named '%s', "
+                        + "but his type does not meet the required. "
+                        + "Expected: %s. Found: %s", name, type.getName(), elementType.getName()));
+            }
+        } else if (!(type.equals(WebElement.class) || isHtmlElement(elementType))) {
+            return (T) createElementWithCustomType(instanсeType, element);
+        }
+        return (T) element;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast to "T". ([squid:S1905](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1905))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=133&issues=AWfPt2o64tUZZzFFj6Ha&open=AWfPt2o64tUZZzFFj6Ha)

[ISSUE_KEY]: AWfPt2o64tUZZzFFj6Ha